docs/adr/adr-005-data-retention.md

```diff
@@ -0,0 +1,201 @@
+lists prevent us from doing simple things easily, like determining which lists
+are actually in use.
+
+[archived]: ...
+[deleted]: ...
+[nullified]: ...
+[initial policy]: https://github.com/alphagov/email-alert-api/blob/master/docs/adr/adr-003-initial-data-retention.md
+[migrate subscribers to daily digest emails]: https://github.com/alphagov/email-alert-api/pull/1352
+
+---
+
+## Decision
+
+The decision we came to was to implement a [worker] which runs periodically
+every day at [midday] to remove data we classify as historic. Removing this
+data was deemed as non urgent and so deleting once a day, in hours, seemed
+sufficient.
+
+To deem data as historic we had to be ensure that the system no longer
+uses the data, and we don't need the data for future analytics.
+
+Below are the decisions we made.
+
+### Max retention of 1 year for everything that's unused by the system
+
+There is no evidence of needing unused data for analytics beyond 1 year and so
+we can classify this old data as historic and remove it.
+
+We can split this unused data into two categories.
+
+Event data - which is only used for a short period after being created, and
+subsequently for analytics, which includes:
+
+* ContentChange & Message - removed once their `created_at` value is over a
+  year old.
+
+* MatchedContentChange & MatchedMessage - cascaded when their parent content
+  change / message is deleted.
+
+* DigestRun - removed once their `created_at` value is over a year old.
+
+* DigestRunSubscriber - cascaded when their parent digest run gets deleted.
+
+Subscriber data - which is used indefinitely until it's marked as "ended", which
+includes:
+
+* Subscription - removed when they've been ended for over a year, based on
+  their `ended_at` value. Once a subscription is ended it [can't currently be
+  made active again] and so can be deleted.
+
+* SubscriberList - removed when they have no active subscriptions and their
+  latest, ended subscription is over a year old.
+
+* Subscriber - removed once their `created_at` is over a year old and they
+  don't have any subscriptions (active or ended).
+
+### Shorter retention of 7 days for subscriber lists with no subscriptions
+
+We also remove subscriber lists which have no subscriptions (active or ended)
+and are over 7 days old. As mentioned in the context section, dead subscriber
+lists pollute our analytics and should be removed regularly.
+
+A scenario of when this can happen is when a user quits their signup journey
+midway through. 7 days was the chosen retention period to allow users to complete
+their signup journey as they have a [7 day limit] to confirm their
+subscription.
+
+### Modified the nullify process and retired the deactivated concept
+
+The [deactivated] concept was a point of confusion in the email system
+and wasn't actually needed so has been removed. It revolved around the
+`deactivated_at` field on the subscriber model. This field was mainly used as a
```

Comment:
```suggestion
`deactivated_at` field on the subscriber model. This field was only used as a
```

Minor: to avoid sounding like we're not sure if we broke something 😄.

Reply:
😆